deepchecks/base/dataset.py

```diff
@@ -126,6 +144,168 @@ def __init__(self, df: pd.DataFrame,
+                "First array must contains the features and second the labels."
+            )
+
+        def validate_features_array(
+            features_array: np.ndarray,
+            feature_names: t.Optional[t.Sequence[t.Hashable]] = None
+        ) -> t.Tuple[np.ndarray, t.List[t.Hashable]]:
+            message = (
+                "'from_numpy' constructor expecting features (args[0]) "
+                "to be not empty two dimensional array."
+            )
+            if len(features_array.shape) != 2:
+                raise ValueError(message)
+            if features_array.shape[0] == 0 or features_array.shape[1] == 0:
+                raise ValueError(message)
+
+            if feature_names is not None and len(feature_names) != features_array.shape[1]:
+                raise ValueError(
+                    f'{features_array.shape[1]} features were provided '
+                    f'but only {len(feature_names)} name(s) for them`s.'
+                )
+            elif feature_names is None:
+                feature_names = [f'X{index}'for index in range(1, features_array.shape[1] + 1)]
+
+            return features_array, t.cast(t.List[str], feature_names)
+
+        if len(args) == 1:
+            features_array, features_names = validate_features_array(args[0], features)
+
+            if label is not None:
+                raise ValueError(
+                    "'label' parameter cannot be passed to the "
+                    "constructor whereas labels array was not provided."
+                )
+
+            return cls(
+                df=pd.DataFrame(data=features_array, columns=features_names),
+                features=features_names, # type: ignore TODO
+                **kwargs
+            )
+
+        else:
+            features_array, features_names  = validate_features_array(args[0], features)
+            labels_array = args[1]
+            label = label or 'target'
+            columns = features_names + [label]
+
+            if len(labels_array.shape) != 1 or labels_array.shape[0] == 0:
+                raise ValueError(
+                    "'from_numpy' constructor expecting labels (args[1]) "
+                    "to be not empty one dimensional array."
+                )
+
+            if labels_array.shape[0] != features_array.shape[0]:
+                raise ValueError(
+                    "'from_numpy' constructor expecting that features and "
+                    "labels arrays will be of the same size"
+                )
+
+            labels_array = labels_array.reshape(len(labels_array), 1)
+            data = np.hstack((features_array, labels_array))
+
+            return cls(
+                df=pd.DataFrame(data=data, columns=columns),
+                features=features_names, # type: ignore TODO
+                label=label, # type: ignore TODO
+                **kwargs
+            )
+
+    @classmethod
+    def from_dict(
```

Comment:
In which use cases from_dict is needed?

Reply:
I thought it could be useful, but if it is not needed I will remove it

Reply:
I think it could be useful